darts/timeseries.py

```diff
@@ -5375,118 +5373,109 @@ def __truediv__(self, other):
                 ),
                 logger,
             )
+        return np.greater(self._values, other)
 
-    def __le__(self, other) -> xr.DataArray:
-        if isinstance(other, (int, float, np.integer, np.ndarray, xr.DataArray)):
-            return _xarray_with_attrs(
-                self._xa <= other, self.static_covariates, self.hierarchy, self.metadata
-            )
-        elif isinstance(other, TimeSeries):
-            return _xarray_with_attrs(
-                self._xa <= other.data_array(copy=False),
-                self.static_covariates,
-                self.hierarchy,
-                self.metadata,
-            )
-        else:
+    def __le__(self, other) -> np.ndarray:
+        if isinstance(other, (TimeSeries, xr.DataArray, np.ndarray)):
+            other = self._extract_values(other)
+        elif not isinstance(other, (int, float, np.integer)):
             raise_log(
                 TypeError(
-                    f"unsupported operand type(s) for < : '{type(self).__name__}' and '{type(other).__name__}'."
+                    f"unsupported operand type(s) for <= : '{type(self).__name__}' and '{type(other).__name__}'."
                 ),
                 logger,
             )
+        return np.less_equal(self._values, other)
 
-    def __ge__(self, other) -> xr.DataArray:
-        if isinstance(other, (int, float, np.integer, np.ndarray, xr.DataArray)):
-            return _xarray_with_attrs(
-                self._xa >= other, self.static_covariates, self.hierarchy, self.metadata
-            )
-        elif isinstance(other, TimeSeries):
-            return _xarray_with_attrs(
-                self._xa >= other.data_array(copy=False),
-                self.static_covariates,
-                self.hierarchy,
-                self.metadata,
-            )
-        else:
+    def __ge__(self, other) -> np.ndarray:
+        if isinstance(other, (TimeSeries, xr.DataArray, np.ndarray)):
+            other = self._extract_values(other)
+        elif not isinstance(other, (int, float, np.integer)):
             raise_log(
                 TypeError(
-                    f"unsupported operand type(s) for < : '{type(self).__name__}' and '{type(other).__name__}'."
+                    f"unsupported operand type(s) for >= : '{type(self).__name__}' and '{type(other).__name__}'."
                 ),
                 logger,
             )
+        return np.greater_equal(self._values, other)
 
     def __str__(self):
-        return str(self._xa).replace("xarray.DataArray", "TimeSeries (DataArray)")
+        return str(self.data_array(copy=False)).replace(
+            "xarray.DataArray", "TimeSeries (DataArray)"
+        )
 
     def __repr__(self):
-        return self._xa.__repr__().replace("xarray.DataArray", "TimeSeries (DataArray)")
+        return (
+            self.data_array(copy=False)
+            .__repr__()
+            .replace("xarray.DataArray", "TimeSeries (DataArray)")
```

Comment:
Do we really want to keep the "(DataArray)" in the representation since it not used under the hood anymore?

Reply:
Removing it sounds good 👍